apps/emqx_bridge_confluent/src/emqx_bridge_confluent_producer.erl

```diff
@@ -0,0 +1,406 @@
+                segment_bytes => <<"100MB">>,
+                memory_overload_protection => true
+            }
+        },
+        local_topic => <<"mqtt/local/topic">>
+    }.
+
+%%-------------------------------------------------------------------------------------------------
+%% `emqx_connector_resource' API
+%%-------------------------------------------------------------------------------------------------
+
+connector_config(Config) ->
+    %% Default port for Confluent is 9092
+    BootstrapHosts0 = maps:get(bootstrap_hosts, Config),
+    BootstrapHosts = emqx_schema:parse_servers(
+        BootstrapHosts0,
+        ?MODULE:host_opts()
+    ),
+    Config#{bootstrap_hosts := BootstrapHosts}.
+
+%%-------------------------------------------------------------------------------------------------
+%% Internal fns
+%%-------------------------------------------------------------------------------------------------
+
+ref(Name) ->
+    hoconsc:ref(?MODULE, Name).
+
+connector_overrides() ->
+    #{
+        authentication =>
+            mk(
+                ref(auth_username_password),
+                #{
+                    default => #{},
+                    required => true,
+                    desc => ?DESC("authentication")
+                }
+            ),
+        bootstrap_hosts =>
+            mk(
+                binary(),
+                #{
+                    required => true,
+                    validator => emqx_schema:servers_validator(
+                        host_opts(), _Required = true
+                    )
+                }
+            ),
+        ssl => mk(
+            ref(ssl_client_opts),
+            #{
+                required => true,
+                default => #{<<"enable">> => true}
+            }
+        ),
+        type => mk(
+            ?CONFLUENT_CONNECTOR_TYPE,
+            #{
+                required => true,
+                desc => ?DESC("connector_type")
+            }
+        )
+    }.
+
+bridge_v2_overrides() ->
+    #{
+        parameters =>
+            mk(ref(producer_kafka_opts), #{
+                required => true,
+                validator => fun emqx_bridge_kafka:producer_strategy_key_validator/1
+            }),
+        ssl => mk(ref(ssl_client_opts), #{
+            default => #{
+                <<"enable">> => true,
+                <<"verify">> => <<"verify_none">>
+            }
+        }),
+        type => mk(
+            ?CONFLUENT_CONNECTOR_TYPE,
+            #{
+                required => true,
+                desc => ?DESC("bridge_v2_type")
+            }
+        )
+    }.
+auth_overrides() ->
+    #{
+        mechanism =>
+            mk(plain, #{
+                required => true,
+                default => plain,
+                importance => ?IMPORTANCE_HIDDEN
+            }),
+        username => mk(binary(), #{required => true}),
+        password => emqx_connector_schema_lib:password_field(#{required => true})
```

Comment:
In the requirement doc:

- Rename "Username" to "Key", make it required
- Rename "Password" to "Secret", make it required

Reply:
Yes, it was done via labels, because it's an UI thing:

https://github.com/emqx/emqx/pull/11989/files#diff-ed0de0f73817dfdfb7325cb0b9a4185da0555b6334bd5601506059f6f3df95feR314

https://github.com/emqx/emqx/pull/11989/files#diff-ed0de0f73817dfdfb7325cb0b9a4185da0555b6334bd5601506059f6f3df95feR320